### DISCUSSION TITLE
How does Sonoff D1 work?

### DISCUSSION THREAD

**Discussion by decimoseptimo** [arendst/Tasmota#11224] [Q&A] [answered]
I have no interest in using stock firmware due to the internet requirement. Neither the DIY mode due requiring the physical remote to enable it. 

What are the capabilities of this device with tasmota firmware? I took a look a the [code](https://github.com/arendst/Tasmota/blob/master/tasmota/xdrv_37_sonoff_d1.ino), but i'm not knowlegeable in tasmota.

**Comment by barbudor** [comment, marked as answer]
Sonoff D1 is a Dimmer.
With Tasmota you can control it without the need for Sonoff Cloud.
Tasmota doesn't need an Internet access however it is not intended to be used standalone a some infrastructure is required :
- An access to an NTP server to provide date reference. If Internet is not available, a local one or a GPS receiver is required
- A MQTT broker is recommended, although not mandatory
- Home automation or backend can take advantage of Tasmota+MQTT (still not mandatory)
Without home automation and MQTT, Tasmota device can still be controlled from its Web GUI
- Power and dimming control
- Timers to triggers rules or power at given time-of-day with possible repetition
...

But most of it can be leveraged with MTT and a home automation

Notet that the "DIY mode" is a method to flash alternative software on a SOnoff device.
Flashing Tasmota requires to either use the DIY mode or to open the unit for serial flashing with an appropriate USB-to-Serial adapter
This is a one-time operation.

**Reply by decimoseptimo** [reply]
Does its WebGUI allow dimming control through LAN? (set and forget) i think openwrt has ntp capabilities. Or do i have to get invested in home automation?

**Reply by barbudor** [reply]
Not sure to understand what you mean by "set and forget"
On the WebGUI you will have a slider that you can adjust and a button to toggle power on/off
You can also control through MQTT but needs a MQTT broker and an backend/frontend (home automation or app)
You can also use web request although this is not the recommended Tasmota usage

**Reply by decimoseptimo** [reply]
I meant, it won't lose its configuration upon restart, which i suppose it won't. 
Thanks mate, have a nice day.

**Reply by barbudor** [reply]
Normally it won't.
